dask/dataframe/multi.py

```diff
@@ -602,6 +615,13 @@ def concat(dfs, axis=0, join='outer', interleave_partitions=False):
                 raise ValueError('All inputs have known divisions which cannot '
                                  'be concatenated in order. Specify '
                                  'interleave_partitions=True to ignore order')
+    elif (axis == 1 and
+          len(dasks) == len(dfs) and
+          all(not df.known_divisions for df in dfs) and
+          len({df.npartitions for df in dasks}) == 1):
+        warn("Concatenating dataframes assuming that they have the same length."
```

Comment:
In this case all dataframes have `divisions == [None, None, None, ...]`.  But yes, perhaps a statement like we're assuming that all of the dataframes are partitioned so that their indices are aligned?